google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -261,6 +272,21 @@ protected String defaultHost() {
   }
 
   protected String defaultProject() {
+    return defaultProjectId();
+  }
+
+  /**
+   * Returns the default project ID, or {@code null} if no default project ID could be found. This
+   * method returns the first available project ID among the following sources:
+   * <ol>
+   *   <li>App Engine project ID
```

Comment:
It seems as though you get it from PROJECT_ENV_NAME, then LEGACY_PROJECT_ENV_NAME